sdk/state/ingest.go

```diff
@@ -15,71 +15,70 @@ import (
-	// mark the channel as closing.
-	// TODO: If the tx hash matches an authorized or unauthorized close, mark the
-	// channel as closed.
-	// TODO: If the tx is for an older declaration, mark the channel as closing with
-	// requiring bump.
-	// TODO: Use the transaction result to affect on success/failure.
+	if closeState == CloseStateEarlyClosing {
+		err := c.updateUnauthorizedAgreement(tx)
+		if err != nil {
+			return err
+		}
+		closeState, err := c.CloseState()
+		if err != nil {
+			return fmt.Errorf("getting close state: %w", err)
+		}
+		if closeState != CloseStateClosing {
+			return fmt.Errorf("converting unauthorized agreement to authorized should put channel in closing state")
+		}
+	}
 
 	return nil
 }
 
-// ingestTxToUpdateUnauthorizedCloseAgreement uses the signatures in the
-// transaction to authorize an unauthorized close agreement if the channel has
-// one. This process helps to give a participant who proposed an agreement the
+func (c *Channel) ingestTxToUpdateInitiatorEscrowAccountSequence(tx *txnbuild.Transaction) {
+	// If the transaction's source account is not the initiator's escrow
+	// account, then the transaction is not a part of a close agreement.
+	if tx.SourceAccount().AccountID != c.initiatorEscrowAccount().Address.Address() {
+		return
+	}
+
+	c.setInitiatorEscrowAccountSequence(tx.SourceAccount().Sequence)
+}
+
+// updateUnauthorizedAgreement uses the signatures in the transaction to
+// authorize an unauthorized close agreement if the channel has one.
+// This process helps to give a participant who proposed an agreement the
 // ability to close the channel if they did not receive the confirmers
 // signatures for a close agreement when the agreement was being negotiated. If
 // the transaction cannot be used to do this the function returns a nil error.
 // If the transaction should be able to provide this data and cannot, the
 // function errors.
-func (c *Channel) ingestTxToUpdateUnauthorizedCloseAgreement(tx *txnbuild.Transaction) error {
-	// If the transaction's source account is not the initiator's escrow
-	// account, then the transaction will not be a transaction for a close
-	// agreement and won't be able to update any unauthorized close agreement.
-	if tx.SourceAccount().AccountID != c.initiatorEscrowAccount().Address.Address() {
-		return nil
+func (c *Channel) updateUnauthorizedAgreement(tx *txnbuild.Transaction) error {
+	if c.latestUnauthorizedCloseAgreement.isEmpty() {
+		return fmt.Errorf("unauthorized close agreement is empty")
 	}
 
 	ca := c.latestUnauthorizedCloseAgreement
 
-	// If there is no unauthorized close agreement, then there's no need to try
-	// and update it.
-	if ca.isEmpty() {
-		return nil
-	}
-
-	// Load the declaration and close transactions for the unauthorized close
-	// agreement.
-	declTx, closeTx, err := c.closeTxs(c.openAgreement.Details, ca.Details)
+	declTx, closeTx, err := c.closeTxs(c.openAgreement.Details, c.latestUnauthorizedCloseAgreement.Details)
```

Comment:
This change appears unnecessary, the close agreement is already loaded above:
```suggestion
	declTx, closeTx, err := c.closeTxs(c.openAgreement.Details, ca.Details)
```

Reply:
changed back here: https://github.com/stellar/experimental-payment-channels/pull/232/commits/ec6870bb18fe64951de8df8800bb495699a072f4